Code/src/main/java/nl/utwente/group10/ui/components/ConnectionStateDependent.java

```diff
@@ -0,0 +1,49 @@
+package nl.utwente.group10.ui.components;
+
+import nl.utwente.group10.ui.handlers.ConnectionCreationManager;
+
+/**
+ * Interface for Nodes who are dependent on their ConnectionState.
+ */
+public interface ConnectionStateDependent {    
+    /**
+     * Tells the Node that its current state (considering connections) possibly has
+     * changed. Default implementation does not react to a potential state
+     * change.
+     *
+     * This method should only be called after the Node's constructor is done.
+     * 
+     * This method will invalidate the Node even if the state did not change.
+     */
+    public void invalidateConnectionState();
+    
+    /**
+     * @return The ConnectionState the Node is in.
+     */
+    public int getConnectionState();
```

Comment:
I wanted to make connectionStateIsUpToDate() a default method.
Java interfaces do not support instance fields, so I solved that by adding a getConnectionState().
Now that I think of it, this method can probably be protected (Does an interface support this?).

Removing this would mean that every Class that implements this interface would have to implement isUpToDate